hooks/node_handlers/base_export_handler.py

```diff
@@ -0,0 +1,644 @@
+        """
+        return self.get_work_template(node)
+    
+    def _populate_from_fields(self, node, fields):
+        """
+        Populate the node from template fields.
+
+        :param node: A :class:`hou.Node` instance.
+        :param dict fields: The template fields.
+        """
+        sgtk_element = node.parm(self.SGTK_ELEMENT)
+        sgtk_element.set(fields.get("name", ""))
+
+        sgtk_location = node.parm(self.SGTK_LOCATION)
+        sgtk_location.set(fields.get("location", ""))
+
+        sgtk_variation = node.parm(self.SGTK_VARIATION)
+        sgtk_variation.set(fields.get("variation", ""))
+
+    def _set_version(self, node, current_version):
+        """
+        Set the index of the versions drop down from the given version.
+
+        :param node: A :class:`hou.Node` instance.
+        :param int current_version: The version to set.
+        """
+        version_str = str(current_version)
+        sgtk_version = node.parm(self.SGTK_VERSION)
+        entries = sgtk_version.menuItems()
+        if version_str not in entries:
+            index = len(entries)
+        else:
+            index = entries.index(version_str)
+        sgtk_version.set(index)
+
+    def _get_optional_fields(self, node, template):
+        """
+        Get the optional fields from the sgtk_optional_keys parm.
+
+        :param node: A :class:`hou.Node` instance.
+        :param template: An :class:`sgtk.Template` instance.
+
+        :rtype: dict
+        """
+        parm = node.parm(self.OPTIONAL_KEYS)
+        if not parm:
+            return {}
+        return json.loads(parm.evalAsString())
+
+    def _populate_from_file_path(self, node, file_path):
+        """
+        Populate a node's sgtk parms from a file path.
+        If the file path doesn't match the template, disable the
+        use of shotgun on this node.
+
+        :param node: A :class:`hou.Node` instance.
+        :param str file_path: The file path to populate from.
+        """
+        template = self._get_template_for_file_path(node, file_path)
+        skip_keys = self.get_optional_keys(template)
+        fields = template.validate_and_get_fields(file_path, skip_keys=skip_keys)
+        if not fields:
+            use_sgtk = node.parm(self.USE_SGTK)
+            use_sgtk.set(False)
+            self._enable_sgtk(node, False)
+            output_parm = node.parm(self.OUTPUT_PARM)
+            output_parm.set(file_path)
+        else:
+            fields.update(self._get_optional_fields(node, template))
+            fields["SEQ"] = "FORMAT: $F"
+            current_version = fields.get("version", self.NEXT_VERSION_STR)
+            all_versions = self._resolve_all_versions_from_fields(node, fields, template)
+            self._populate_from_fields(node, fields)
+            self._update_all_versions(node, all_versions)
+            self._set_version(node, current_version)
+            self._refresh_file_path(node, update_version=False)
```

Comment:
Switch it around to get rid of extra `not`:
```suggestion
        if fields:
            fields.update(self._get_optional_fields(node, template))
            fields["SEQ"] = "FORMAT: $F"
            current_version = fields.get("version", self.NEXT_VERSION_STR)
            all_versions = self._resolve_all_versions_from_fields(node, fields, template)
            self._populate_from_fields(node, fields)
            self._update_all_versions(node, all_versions)
            self._set_version(node, current_version)
            self._refresh_file_path(node, update_version=False)
        else:
            use_sgtk = node.parm(self.USE_SGTK)
            use_sgtk.set(False)
            self._enable_sgtk(node, False)
            output_parm = node.parm(self.OUTPUT_PARM)
            output_parm.set(file_path)
```